src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java

```diff
@@ -1771,6 +1773,58 @@ public Response linkDataverse(@Context ContainerRequestContext crc, @PathParam("
             return ex.getResponse();
         }
     }
+    
+    @GET
+    @AuthRequired
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{identifier}/{type}/linkingDataverses")
+    public Response getLinkingDataverseList(@Context ContainerRequestContext crc, @PathParam("identifier") String dvIdtf, @QueryParam("searchTerm") String searchTerm, @PathParam("type") String type) {
+        //first determine what you are linking based on identifier and type
+
+        try {
+
+            DvObject dvObject = findDvoByIdAndTypeOrDie(dvIdtf, type, false);
+            List<Dataverse> dataversesForLinkingSearch = new ArrayList();
+            
+            AuthenticatedUser requestUser = (AuthenticatedUser) getRequestUser(crc);
+            DataverseRequest dvReq = new DataverseRequest(requestUser, (IpAddress) null);
+            List<Dataverse> dataversesForLinking;
+            dataversesForLinking = permissionService.findPermittedCollections(dvReq, requestUser, Permission.LinkDataset);
+
+            List<Dataverse> mergedWithSearch = new ArrayList<>();
+            dataversesForLinking = dataverseService.removeUnlinkableDataverses(dataversesForLinking, dvObject);
+            
+            //Only do search lookup if search term is there. Otherwise just include the collections based on perms
+            if (!searchTerm.isEmpty()) {
+                dataversesForLinkingSearch = dataverseService.filterDataversesByNamePattern(searchTerm);
+                if (!dataversesForLinkingSearch.isEmpty()) {
+                    for (Dataverse dv : dataversesForLinking) {
+                        if (dataversesForLinkingSearch.contains(dv)) {
+                            mergedWithSearch.add(dv);
+                        }
+                    }
+                }
+            } else {
+                //search term empty then add all based on perms
+                mergedWithSearch.addAll(dataversesForLinking);
+            }
+            
+            JsonArrayBuilder dvBuilder = Json.createArrayBuilder();
+            if (!mergedWithSearch.isEmpty()) {
+                for (Dataverse dv : mergedWithSearch) {
+                    dvBuilder.add(json(dv, true));
+                }
+            }
```

Comment:
JsonPrinter should be fully responsible of building the response, given the entire list.